py/vm.c

```diff
@@ -61,8 +61,30 @@
     do { \
         unum = (unum << 7) + (*ip & 0x7f); \
     } while ((*ip++ & 0x80) != 0)
-#define DECODE_ULABEL size_t ulab = (ip[0] | (ip[1] << 8)); ip += 2
-#define DECODE_SLABEL size_t slab = (ip[0] | (ip[1] << 8)) - 0x8000; ip += 2
+
+#define DECODE_ULABEL \
```

Comment:
Are these complex enough that they should be functions now?

Reply:
They are not that complicated when converted to machine instructions.  In this case I'm aiming for performance of the VM rather than size.  Making them functions, I'm not sure if the compiler would inline them or not.  Maybe in small builds with `-Os` we do want them to be functions, but it'd slow down the VM (maybe).

Also, making them functions requires passing in `ip` as a pointer, ie `&ip`, so it can be updated.  That can significantly change things in the VM code generation because `ip` is intended to be a local register in a variable.  For that reason alone I'd prefer to keep these as macros.